cfme/tests/insights/test_rhi_inventory.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+
+from cfme import test_requirements
+from cfme.configure.configuration.analysis_profile import AnalysisProfile
+from cfme.configure.configuration.region_settings import RedHatUpdates
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.markers.env_markers.provider import ONE_PER_VERSION
+from cfme.utils import conf, ssh
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.conf import credentials
+from cfme.utils.hosts import setup_host_creds
+from cfme.utils.log import logger
+from cfme.utils.wait import wait_for, wait_for_decorator
+
+
+pytestmark = [
+    test_requirements.insights,
+    pytest.mark.meta(server_roles="+smartproxy +smartstate"),
+    pytest.mark.provider([VMwareProvider], selector=ONE_PER_VERSION, required_fields=[
+        ['vm_analysis_new', 'provisioning', 'vlan'],
+        ['vm_analysis_new', 'provisioning', 'host'],
+        ['vm_analysis_new', 'provisioning', 'datastore']]),
+    pytest.mark.usefixtures('setup_provider')
+]
+
+
+@pytest.fixture
+def register_appliance(request, appliance, reg_method='rhsm'):
+    red_hat_updates = RedHatUpdates(service='rhsm', url='subscription.rhn.redhat.com',
+                                    username=conf.credentials[reg_method]['username'],
+                                    password=conf.credentials[reg_method]['password'])
+
+    if not red_hat_updates.is_registered:
+        red_hat_updates.update_registration()
+
+        red_hat_updates.register_appliances()
+        wait_for(
+            func=red_hat_updates.is_registering,
+            func_args=[appliance.server.name],
+            delay=10,
+            num_sec=300,
+            fail_func=red_hat_updates.refresh
+        )
```

Comment:
**REQUIRED**
Why do we need this `wait_for`? Below you already have one which waits for `red_hat_updates.is_registered`